summary/summary.py

```diff
@@ -30,21 +31,77 @@
                       "Pass a list of scids to the {exclude} parameter"\
                       " to exclude some channels from the outputs."
 
+# Global state to measure online% and last_seen
+peerstate = {}
+
+
+# ensure an rpc peer is added
+def addpeer(p):
+    pid = p['id']
+    if not pid in peerstate:
+        peerstate[pid] = {
+            'connected' : p['connected'],
+            'last_seen' : datetime.now() if p['connected'] else None,
+            'availability' : 1.0 if p['connected'] else 0.0
+        }
+
+
+class PeerThread(threading.Thread):
+    def __init__(self):
+        super().__init__()
+        self.daemon = True
+        self.start()
+
+    def run(self):
+        interval = 5 * 60            # collect peer state once in a while
+        window   = 3 * 24 * 60 * 60  # 72hr hr availability window
+        count    = 0
+
+        # delay initial execution, so peers have a chance to connect on startup
+        time.sleep(interval)
+
+        while True:
+            count += 1
+            leadwin  = max(min(window, count * interval), count * interval)
+            samples  = leadwin / interval
+            alpha    = 1.0 - 1.0 / samples
+            beta     = 1.0 - alpha
+
+            try:
+                peers = plugin.rpc.listpeers()
+                for p in peers['peers']:
+                    pid = p['id']
+                    addpeer(p)
+
+                    if p['connected']:
+                        peerstate[pid]['last_seen'] = datetime.now()
+                        peerstate[pid]['connected'] = True
+                        peerstate[pid]['availability'] = peerstate[pid]['availability'] * alpha + 1.0 * beta
+                    else:
+                        peerstate[pid]['connected'] = False
+                        peerstate[pid]['availability'] = peerstate[pid]['availability'] * alpha + 0.0 * beta
```

Comment:
i've never seen this alpha/beta trick before, does it have a name?

Reply:
exponential smoothing

Reply:
and yes, `0.0 * beta` is pointless, but it would also work for values in between, which do not happen for bolean states :D

Reply:
so if i understand this simple exponential smoothing algo from [wikipedia](https://en.wikipedia.org/wiki/Exponential_smoothing)

![exponential smoothing](https://wikimedia.org/api/rest_v1/media/math/render/svg/0248d7cdd017b01a3c1d29d857915f1e7a2eac04)

What this would translate to in this case is 

```
alpha * (if p['connected'] 1.0 else 0.0) + beta * peerstate[pid]['availability']
```

- `alpha` is multiplied by the input at this window, in this case 'is_connected'
- `beta` is multiplied by the previous output, or `peerstate[pid]['availability']`

Reply:
looking at this more closely, your `alpha` and `beta` calculations are flipped with respect to the wikipedia example -- as is you've got it coded correctly.

Reply:
I flipped alpha / beta to make it more regular :D